From 056ee33508fad36e160d6dc93ca165ed0fd36841 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 17 Oct 2022 19:40:37 +0300
Subject: [PATCH] [SW-0] gaudi3: store fuse values before slv fuse emulation init

Currently, fuse values of master die in case of fuse emulation is stored
in global data structure after slave fuse emulation init. Due to
addition of spi gateway fuse bit check while spi init of fuse emulation,
this patch moves assignment of master fuse values in global data
structure before slave fuse emulation init.

Change-Id: Ic9e0daf3647bb44dc4ff051362821119ba3f96a3
---

diff --git a/arch/gaudi3/arc/spi_flash.c b/arch/gaudi3/arc/spi_flash.c
index ad74551..7526561 100644
--- a/arch/gaudi3/arc/spi_flash.c
+++ b/arch/gaudi3/arc/spi_flash.c
@@ -88,7 +88,7 @@
 {
 	struct gaudi3_cold_rst_data cold_rst_data;
 	struct boot_desc desc;
-	uint32_t val, idx;
+	uint32_t idx;
 
 	if (arcboot_rom_global.die_id) {
 		arcboot_rom_wait_for_fuse_data();
@@ -105,13 +105,10 @@
 	}
 	arcboot_rom_global.fuse_spi = true;
 
-	if (fuse_read(EFUSE_SYSCFG_OFFSET, &val) == -1)
-		val = EFUSE_SYSCFG_DEFAULT;
+	fuse_read(EFUSE_SYSCFG_OFFSET, &arcboot_rom_global.config_fuse);
+	fuse_read(EFUSE_SYSCFG2_OFFSET, &arcboot_rom_global.config2_fuse);
 
 	spi_flash_slave_init();
-
-	arcboot_rom_global.config_fuse = val;
-	fuse_read(EFUSE_SYSCFG2_OFFSET, &arcboot_rom_global.config2_fuse);
 
 	/* Since spi fuse init is successful, set fuse_emulation_done to true */
 	if (!arcboot_rom_global.fuse_emulation_done) {
